pkg/watchers/watcher.go

```diff
@@ -28,7 +29,8 @@ var (
 type Watcher struct {
 	ctrl             controller.Controller
 	cache            cache.Cache
-	watched          map[string]interface{}
+	watches          map[string]bool
+	wmut             sync.RWMutex
```

Comment:
why not just use https://pkg.go.dev/golang.org/x/sync/syncmap ?

Reply:
I actually started with sync.Map but this isn't type safe (works with `any`) and the API is less convenient than the typical map API .. since I'm only using the mutex in a couple of places, all in all I find it easier to use traditional map + mutex